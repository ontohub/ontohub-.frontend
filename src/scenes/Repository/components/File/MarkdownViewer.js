import React, { Fragment } from "react";
import { Container } from "semantic-ui-react";
import Markdown from "react-markdown";
import _githubMarkdownCss from "github-markdown-css/github-markdown.css";

export const MarkdownViewer = ({ value }) => (
  // We assume that the value is encoded in plain text
  <Container className="markdown-body">
    <Markdown escapeHtml={true} source={value} />
  </Container>
);
