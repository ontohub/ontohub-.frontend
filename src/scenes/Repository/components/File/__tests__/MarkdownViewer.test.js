import { MarkdownViewer } from "../MarkdownViewer";

describe("MarkdownViewer", () => {
  set("component", () => (
    <MarkdownViewer
      value={`
# Hello [World](https://ontohub.org)

\`\`\`
() => null;
\`\`\`
`}
    />
  ));
  set("renderedComponent", () => render(component));

  it("matches the snapshot", () => {
    expect(renderedComponent).toMatchSnapshot();
  });
});
