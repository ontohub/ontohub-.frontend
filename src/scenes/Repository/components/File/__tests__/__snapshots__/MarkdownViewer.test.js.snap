// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarkdownViewer matches the snapshot 1`] = `
<div
  class="ui container markdown-body"
>
  <div>
    <h1>
      Hello 
      <a
        href="https://ontohub.org"
      >
        World
      </a>
    </h1>
    <pre
      class="lowlight"
    >
      <code
        class="hljs yaml"
      >
        <span
          class="hljs-string"
        >
          ()
        </span>
         
        <span
          class="hljs-string"
        >
          =&gt;
        </span>
         
        <span
          class="hljs-literal"
        >
          null
        </span>
        <span
          class="hljs-string"
        >
          ;
        </span>
      </code>
    </pre>
  </div>
</div>
`;
